src/native/minipal/cpufeatures.c

```diff
@@ -441,19 +450,18 @@ int minipal_getcpufeatures(void)
     return result;
 }
 
-// Detect if the current process is running under the Apple Rosetta x64 emulator
-bool minipal_detect_rosetta(void)
+static const char* GetCpuBrand(void)
 {
 #if defined(HOST_AMD64) || defined(HOST_X86)
     // Check for CPU brand indicating emulation
     int regs[4];
-    char brand[49];
+    static char brand[49];
```

Comment:
This should be buffer passed in by the caller. I do not want to think about the race conditions that this may possibly have.